core/vm/evm.go

```diff
@@ -416,6 +437,9 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 	}
 	nonce := evm.StateDB.GetNonce(caller.Address())
 	evm.StateDB.SetNonce(caller.Address(), nonce+1)
+	// We add this to the access list _before_ taking a snapshot. Even if the creation fails,
+	// the access-list change should not be rolled back
+	evm.StateDB.AddAccessListAccount(address)
```

Comment:
Should we special case this to only run if YoloV2 is enabled (like `core.ApplyTrsancation)`? Seems saner to me.